yaksh/evaluator_tests/test_c_cpp_evaluation.py

```diff
@@ -52,5 +55,264 @@ def test_infinite_loop(self):
+
+    def test_string_input(self):
+        self.test_case_data = [{'expected_output': 'abc',
+                                'expected_input': 'abc'}]
+        user_answer = dedent("""
+        #include<iostream>
+        using namespace std;
+        int main(void){
+        char a[4];
+        cin>>a;
+        cout<<a;
+        }""")
+        get_class = CppStdioEvaluator()
+        kwargs = {'user_answer': user_answer,
+                  'test_case_data': self.test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertEquals(result.get('error'), "Correct Answer")
+        self.assertTrue(result.get('success'))
+
+    def test_incorrect_answer(self):
+        user_answer = dedent("""
+        #include<iostream>
+        using namespace std;
+        int main(void){
+        int a=10;
+        cout<<a;
+        }""")
+        get_class = CppStdioEvaluator()
+        kwargs = {'user_answer': user_answer,
+                  'test_case_data': self.test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertFalse(result.get('success'))
+        self.assertIn("Incorrect", result.get('error'))
+        self.assertTrue(result.get('error').splitlines > 1)
+
+    def test_error(self):
+        user_answer = dedent("""
+        #include<iostream>
+        using namespace std;
+        int main(void){
+        int a=10;
+        cout<<a
+        }""")
+        get_class = CppStdioEvaluator()
+        kwargs = {'user_answer': user_answer,
+                  'test_case_data': self.test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertFalse(result.get("success"))
+        self.assertTrue("Compilation Error" in result.get("error"))
+
+    def test_infinite_loop(self):
+        user_answer = dedent("""
+        #include<iostream>
+        using namespace std;
+        int main(void){
+        while(0==0){
+        cout<<"abc";}
+        }""")
+        get_class = CppStdioEvaluator()
+        kwargs = {'user_answer': user_answer,
+                  'test_case_data': self.test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertFalse(result.get("success"))
+        self.assertEquals(result.get("error"), self.timeout_msg)
+
+    def test_only_stdout(self):
+        self.test_case_data = [{'expected_output': '11',
+                               'expected_input': ''}]
+        user_answer = dedent("""
+        #include<iostream>
+        using namespace std;
+        int main(void){
+        int a=5,b=6;
+        cout<<a+b;
+        }""")
+        get_class = CppStdioEvaluator()
+        kwargs = {'user_answer': user_answer,
+                  'test_case_data': self.test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertEquals(result.get('error'), "Correct Answer")
+        self.assertTrue(result.get('success'))
+
```

Comment:
To test c and cpp we have two different classes, performing same function, only the answer code changes, can we have a single class to test both?